include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -56,10 +56,14 @@ __device_info(const _ExecutionPolicy& __policy)
 #endif
 
 template <typename _ExecutionPolicy>
-::std::size_t
-__max_work_group_size(const _ExecutionPolicy& __policy)
-{
-    return __policy.queue().get_device().template get_info<sycl::info::device::max_work_group_size>();
+std::size_t
+__max_work_group_size(const _ExecutionPolicy& __policy, std::size_t __wg_size_limit = 8192)
+{
+    std::size_t __wg_size = __policy.queue().get_device().template get_info<sycl::info::device::max_work_group_size>();
+    // Limit the maximum work-group size supported by the device to optimize the throughput or minimize communication
+    // costs. This is limited to 8192 which is the highest current limit of the tested hardware (opencl:cpu devices) to
+    // prevent huge work-group sizes returned on some devices (e.g., FPGU emulation).
+    return std::min(__wg_size, __wg_size_limit);
```

Comment:
Does it make sense to look at the `__kernel_work_group_size` function as well?  It uses a somewhat arbitrary division by 4 for CPU devices.  I think a maximum cap similar to here might be a better system, and it would be more consistent.

Reply:
Good point. I've removed this arbitrary value and used specific limits as discussed above.

Reply:
Thanks again for looking at this.  

I wonder if we should continue to restrict this cap to CPU devices (for the caps which are replacing the CPU specific division by 4).  In practice, with current hardware and drivers, I believe 1024 is our GPU cap anyway, but we have no guarantees this does not extend further to 2048 and beyond, and we would not necessarily remember to update these values accordingly.  Presumably, if those sizes are available on the GPU, they will be advantageous to use.

Reply:
I agree that this cap might change with future GPUs, but wouldn't we need to tune for their changed capabilities anyways? It looks to me like these issues might be better addressed by architecture-specific tuning parameters that are stored somewhere centrally and updated for new architectures.

Reply:
Yes, a central location or system for these empirical values and tools to re-evaluate them within some range and options to have specialized values for different hardware would be ideal.  My guess is we don't want to try to solve that in this PR.  

Perhaps for now we can tag these with some specific keyword in the comments and create a github issue which describes how to find these empirical values in the code to later re-evaluate within something like #1054.

The comments you've added describing that these have been empirically determined for CPU are good.  If we don't want to limit with a branch specific to CPU devices, perhaps we can add to that comment saying something like:

// This value exceeds the current practical limit for GPUs, but may need to be re-evaluated in the future

Reply:
I agree that this would be another PR.

I've added comments about re-evaluating future GPUs as you suggested. I prefer this solution over a CPU-specific cap since the CPU and FPGA emulator are currently the only devices using larger work-group sizes making this cap generally applicable.